### PR TITLE
Fix Adaptive Icon support

### PR DESCRIPTION
**Pull request by MariusVolkhart**
### First time contributor checklist

- [x] I have read [how to contribute](https://github.com/signalapp/Signal-Android/blob/master/CONTRIBUTING.md) to this project
- [x] I have signed the [Contributor License Agreement](https://whispersystems.org/cla/)

### Contributor checklist

- [x] I am following the [Code Style Guidelines](https://github.com/signalapp/Signal-Android/wiki/Code-Style-Guidelines)
- [x] I have tested my contribution on these devices:
 * Android emulator, API 27
- [x] My contribution is fully baked and ready to be merged as is
- [x] I ensure that all the open issues my contribution fixes are mentioned in the commit message of my first commit using the `Fixes #1234` [syntax](https://help.github.com/articles/closing-issues-via-commit-messages/)

----------

### Description
Fix Adaptive Icon support

Adaptive Icons were not rendering properly. The circular icon was always being used if the device defined support for it. Additionally, the PNG launcher icons are not appropriately sized for Adaptive Icons.

To remedy this, a new alias is introduced so that on API 26, the Adaptive Icon file is used to render the launcher icon, regardless of what the device defines. This is as prescribed by Google Developer Advocate Ian Lake https://medium.com/@ianhlake/vectordrawable-adaptive-icons-3fed3d3205b5#3a57

A new vector drawable is introduced as the foreground layer of the Adaptive Icon. The source for this drawable is https://github.com/signalapp/Signal-Design/blob/61c5342a0bd1184eaded49685a0cfcf5ec8882d4/android/icons/app_icon/android_material.eps

`colors.xml` was using a different hue of the primary blue than the eps file, so it has been updated with the eps file hue.

Testing was performed using the [AdaptiveIconPlayground](https://github.com/nickbutcher/AdaptiveIconPlayground/releases). The following are examples of how the icon renders.
screen shot 2018-04-21 at 23 53 42
screen shot 2018-04-21 at 23 53 25
screen shot 2018-04-21 at 23 53 03
screen shot 2018-04-21 at 23 52 34

### CODE REVIEW

**Comment by MariusVolkhart** [comment]
To get the VectorDrawable, I made changes to the `android-material.eps` file I based this off of, such as extending the shadow to the edges. I also produced an SVG as that is what Android Studio can convert to a VectorDrawable. Do you want these assets uploaded? If so, is a particular naming convention or folder hierarchy or anything else desired?

File: res/values/colors.xml [hunk 1]
```diff
@@ -1,9 +1,9 @@
 <?xml version='1.0' encoding='UTF-8'?>
 <resources>
-    <color name="signal_primary">#ff2090ea</color>
+    <color name="signal_primary">#ff2190e9</color>
```
Comment (moxie0) [inline res/values/colors.xml]: We can't really change these primary colors. 2090ea is the "signal blue" across platforms

**Comment by MariusVolkhart** [comment]
Reverted the change to `colors.xml`

**Comment by cavega** [comment]
I wonder what is holding this from getting merged. I recall seeing Adaptive Icons mentioned in one of the Play Store releases and to my dismay it didn't actually work. @MariusVolkhart did you use Android Studio's Image Asset tool? I don't see the "Teardrop" shape in your screenshots so just checking that all supported shaped are included in the PR :)

**Comment by greyson-signal** [comment]
Fixed in #7867

**Comment by greyson-signal** [comment]
Fixed in #7867

**Pull request closed without merging**